src/V3HierBlock.cpp

```diff
@@ -0,0 +1,132 @@
+#include <sstream>
+
+#include "V3Ast.h"
+#include "V3Error.h"
+#include "V3HierBlock.h"
+
+//######################################################################
+//
+V3HierBlock::~V3HierBlock() {
+    UASSERT(m_usees.empty(), "at least one module must be a leaf");
+    for (std::set<V3HierBlock*>::const_iterator hit = m_users.begin(); hit != m_users.end();
+         ++hit) {
+        const bool deleted = (*hit)->m_usees.erase(this);
+        UASSERT_OBJ(deleted, m_modp, " is not registered");
+    }
+}
+
+std::string V3HierBlock::Dump() const {
+    std::stringstream ss;
+    ss << "Module:" << m_modp->prettyNameQ() << "\n";
+    for (std::set<const V3HierBlock*>::const_iterator it = m_usees.begin(); it != m_usees.end();
+         ++it) {
+        ss << "    " << (*it)->m_modp->prettyNameQ() << "\n";
+    }
+    return ss.str();
```

Comment:
Maybe this should be a debug file created with --debug like the .dot and some .txt files?  Not sure how often will be referenced.